### ISSUE TITLE
Voice Commands

### ISSUE THREAD

**Issue by WailB** [quran/quran_android#867]
I would like to share a code that I made to play quran audio files using voice commands. This is helpful for people who are having difficulty reading or have eye problems. I hope you will use this code to add this as a feature in your application.﻿

[https://github.com/WailB/QuranByVoice](url)

**Comment by ozbek** [comment]
That's an awesome idea. However, everything seems to be hard-coded in your project. I think, it'd be very hard, if not impossible, to re-use the code in this application. Nevertheless, the idea is very good.

**Comment by WailB** [comment]
I dont think it will be impossible. It will require some work but the basic concept should be the same. Most of the work which is the database for comparing voice commands such as sora names is already made in the arrays. Other code for executing voice commands can be modulized with a little work. I have to clarify that I did not clean the code from other irrelevant code I had from other application so the code might seem bigger than it is actually is.

**Comment by ahmedre** [comment]
i am not against this as long as it doesn't add any new non-runtime permissions, and as long as it's easy to keep off so as to not trigger it by accident. what will the ui look like (if any), and are the commands you have listed in your repository only for audio playback?
jazakAllah khairan

**Comment by WailB** [comment]
The UI is simple only a button to say the voice commands. The app is only proof of concept so it is very basic but functional. It uses google voice commands to convert speech to text. So the commands are only the ones in the readme file. It is easily expandable. You can build the app and check it out yourself. I have added all the files (including images and audio instruction) in the repository except for the actual quran audio files. 

As for how to implement it, I will leave that to you guys to come up with the best way.

**Comment by WailB** [comment]
I have added a working voice commands function to Quran Android app which can be expanded to add additional commands such as audio play or open translation using voice commands.

However, I can't seem to be able to pull the changes to the repo because of permissions.

**Comment by ozbek** [comment]
Please see "Collaborating with issues and pull requests" help page: https://help.github.com/categories/collaborating-with-issues-and-pull-requests/

**Comment by WailB** [comment]
I still can't upload the changes to the repo. I am using github desktop and when I try to sync or publish it says I don't have permissions. Can you be more specific in your response? The link is just referring me to a number of links but I can't seem to find any solution in them.

**Comment by WailB** [comment]
Never mind. I managed to fork a copy and do a pull request on the fork. It is currently pending for your review. Let me know how it goes.
